### PR TITLE
Feature/tl2023 237 review changes to assessment entry UI remove

### PR DESCRIPTION
**Pull request by sajidmalik**

### CODE REVIEW

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs [hunk 1]
```diff
@@ -560,7 +560,36 @@ public async Task<IActionResult> RemoveAssessmentEntryCoreAsync(AdminRemovePathw
             }
 
             await _cacheService.SetAsync(CacheKey, model);
-            return RedirectToRoute(RouteConstants.PageNotFound); // TODO: Redirect to review page
+            return RedirectToRoute(RouteConstants.AdminReviewRemoveCoreAssessmentEntry, new { registrationPathwayId = model.RegistrationPathwayId });
+        }
+
+        [HttpGet]
+        [Route("admin/review-remove-assessment-entry-core/{registrationPathwayId}", Name = RouteConstants.AdminReviewRemoveCoreAssessmentEntry)]
+        public async Task<IActionResult> AdminReviewRemoveCoreAssessmentEntryAsync(int registrationPathwayId)
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs:568]: The parameter is not used at all in the method

File: src/Sfa.Tl.ResultsAndCertification.Web/ViewModel/AdminDashboard/Assessment/AdminCoreComponentViewModel.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@ public class AdminCoreComponentViewModel : AdminAssessmentLearnerDetails
     {
         public AdminAssessmentDetailsViewModel AssessmentDetails { get; set; }
 
-        public IEnumerable<PathwayAssessments> ValidPathwayAssessmentSeries { get; set; }
+        public IList<PathwayAssessments> ValidPathwayAssessmentSeries { get; set; }
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/ViewModel/AdminDashboard/Assessment/AdminCoreComponentViewModel.cs:14]: Why this change? No method of IList is used anywhere

File: src/Sfa.Tl.ResultsAndCertification.Web/ViewModel/AdminDashboard/LearnerRecord/AdminAssessmentDetailsViewModel.cs [hunk 3]
```diff
@@ -14,15 +14,15 @@ public class AdminAssessmentDetailsViewModel
 
         public bool HasAssessmentEntries => !PathwayAssessments.IsNullOrEmpty();
 
-        public IEnumerable<AdminAssessmentViewModel> PathwayAssessments { get; set; }
+        public IList<AdminAssessmentViewModel> PathwayAssessments { get; set; }
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/ViewModel/AdminDashboard/LearnerRecord/AdminAssessmentDetailsViewModel.cs:17]: Why? (Same for specialismDetails)

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs (hunk 1 above)
Comment (sajidmalik) [reply src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs]: Updated

File: src/Sfa.Tl.ResultsAndCertification.Web/ViewModel/AdminDashboard/Assessment/AdminCoreComponentViewModel.cs (hunk 2 above)
Comment (sajidmalik) [reply src/Sfa.Tl.ResultsAndCertification.Web/ViewModel/AdminDashboard/Assessment/AdminCoreComponentViewModel.cs]: Updated

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs [hunk 4]
```diff
@@ -560,7 +560,36 @@ public async Task<IActionResult> RemoveAssessmentEntryCoreAsync(AdminRemovePathw
             }
 
             await _cacheService.SetAsync(CacheKey, model);
-            return RedirectToRoute(RouteConstants.PageNotFound); // TODO: Redirect to review page
+            return RedirectToRoute(RouteConstants.AdminReviewRemoveCoreAssessmentEntry);
+        }
+
+        [HttpGet]
+        [Route("admin/review-remove-assessment-entry-core/{registrationPathwayId}", Name = RouteConstants.AdminReviewRemoveCoreAssessmentEntry)]
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs:567]: You have to remove it from the route too

**Review by ibai-gonzalez — APPROVED** [review]
